unison-core/src/Unison/Name.hs

```diff
@@ -177,10 +180,23 @@ endsWithSegments name ss =
 -- | Like 'endsWithSegments', but accepts a list of name segments in reverse order.
 --
 -- Slightly more efficient than 'endsWithSegments'.
+--
+-- >>> endsWithReverseSegments "a.b.c" ["c", "b"]
+-- True
 endsWithReverseSegments :: Name -> [NameSegment] -> Bool
 endsWithReverseSegments (Name _ ss0) ss1 =
   List.NonEmpty.isPrefixOf ss1 ss0
 
+-- >>> stripReversedPrefix "a.b.c" ["b", "a"]
+-- Just c
+-- >>> stripReversedPrefix "a.b" ["b", "a"]
+-- Nothing
+stripReversedPrefix :: Name -> [NameSegment] -> Maybe Name
```

Comment:
Might be worth showing an example where a mismatching prefix returns `Nothing`, not `Just` the original name.

Reply:
Done 👍🏼 , I love that HLS can update these for me now ❤️